run_flax_speech_recognition_seq2seq_streaming_review.py

```diff
@@ -0,0 +1,1391 @@
+    )
+    model_revision: str = field(
+        default="main",
+        metadata={
+            "help": "The specific model version to use (can be a branch name, tag name or commit id)."},
+    )
+    use_auth_token: bool = field(
+        default=False,
+        metadata={
+            "help": "Will use the token generated when running `transformers-cli login` (necessary to use this script "
+            "with private models)."
+        },
+    )
+    dtype: Optional[str] = field(
+        default="float32",
+        metadata={
+            "help": (
+                "Floating-point format in which the model weights should be initialized and trained. Choose one of"
+                " `[float32, float16, bfloat16]`."
+            )
+        },
+    )
+    num_beams: Optional[int] = field(
+        default=None,
+        metadata={
+            "help": (
+                "Number of beams to use for evaluation. This argument will be passed to `model.generate`, "
+                "which is used during evaluation."
+            )
+        },
+    )
+    dropout: Optional[float] = field(
+        default=None, metadata={"help": "The dropout ratio for the dropout layer probabilities."}
+    )
+    attention_dropout: Optional[float] = field(
+        default=None, metadata={"help": "The dropout ratio for the attention probabilities."}
+    )
+    activation_dropout: Optional[float] = field(
+        default=None, metadata={"help": "The dropout ratio for activations inside the fully connected layer."}
+    )
+    encoder_dropout: Optional[float] = field(
+        default=None, metadata={"help": "The dropout ratio for the encoder layer dropout probabilities."}
+    )
+    decoder_dropout: Optional[float] = field(
+        default=None, metadata={"help": "The dropout ratio for the decoder layer dropout probabilities."}
+    )
+
+@flax.struct.dataclass
+class DataTrainingArguments:
+    """
+    Arguments pertaining to what data we are going to input our model for training and eval.
+    """
+
+    dataset_name: str = field(
+        default=None, metadata={"help": "The name of the dataset to use (via the datasets library)."}
+    )
+    dataset_config_name: Optional[str] = field(
+        default=None, metadata={"help": "The configuration name of the dataset to use (via the datasets library)."}
+    )
+    text_column: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "The name of the column in the datasets containing the full texts (for summarization)."},
+    )
+    dataset_cache_dir: Optional[str] = field(
+        default=None, metadata={"help": "Path to cache directory for saving and loading datasets"}
+    )
+    overwrite_cache: bool = field(
+        default=False, metadata={"help": "Overwrite the cached training and evaluation sets"}
+    )
+    preprocessing_num_workers: Optional[int] = field(
+        default=50,
```

Comment:
The reason we changed the default is that after some experimenting we discover that it was really a waste of compute when training on TPUs, which is our main use case. We believe a similar trend happens on GPUs. We could maybe set it to 8? Any machine with an accelerator device might have at least 8 cores, and the speed benefit really is worth it.

![image](https://user-images.githubusercontent.com/173537/234553427-df5a925a-a406-485b-bbb2-58d81d6b565e.png)

Reply:
That's a crazy speed-up from bumping up the number of workers! On TPU we're almost always limited by our data loading speed (rather than TPU compute speed), so this makes sense from a practical perspective